src/engine/rendering/Model.cpp

```diff
@@ -301,4 +315,43 @@ namespace lei3d
 		return textureID;
 	}
 
+	unsigned int TextureFromMemory(const aiTexture* texture, const std::string& directory) {
+		unsigned int textureID;
+		glGenTextures(1, &textureID);
+
+		bool isCompressed = texture->mHeight == 0;
+		int buflen = isCompressed ? texture->mWidth : texture->mWidth * texture->mHeight * 4;	// TODO: hopefully only "compressed" data, haven't checked the else segment
```

Comment:
yeah good to internalize that this is here though if we start to get real wonky errors n stuff

Reply:
Just for documentation's sake and if this becomes problem later on, this works with the default blender image format, which is PNG. Refer to [here](https://assimp-docs.readthedocs.io/en/latest/usage/use_the_lib.html#textures) on how to deal with embedded textures that may be compressed. Not sure if stb_image will deal with it well.